README.md

```diff
@@ -208,6 +213,183 @@ try {
 
 <a name="help"/>
 
+<a name="in-process-linking"/>
+
+## Trace in-process asynchronous execution
+
+You can use the SDK to trace asynchronous in-process code execution. This might be useful if Dynatrace does not support the threading framework or specific asynchronous libraries. In-process-linking will only be used to link other services (Database-, Webrequests, ...) between thread or queuing boundaries, where currently no support is available.
+
+To link asynchronous execution, you need to receive a InProcessLink, where async execution forks:
+
+```Java
+InProcessLink inProcessLink = OneAgentSDK.createInProcessLink();
+```
+
+The provided inProcessLink must not serialized and can only be used inside the process, where it has been created. It must be used to start tracing where the async execution takes place:
+
+```Java
+InProcessLinkTracer tracer = oneAgentSDK.traceInProcessLink(inProcessLink);
+
+tracer.start();
+try {
+	// do the asynchronous job
+} catch (Exception e) {
+	tracer.error(e);
+} finally {
+	tracer.end();
+}
+```
+
+<a name="messaging"/>
+
+## Trace messages
+
+You can use the SDK to trace messages sent or received via messaging system. When tracing messages, we distinct between:
+* sending a message
+* receiving a message
+* processing a received message
+
+To trace an outgoing message, the code looks straight forward compared to other tracers:
+ 
+```Java
+MessagingSystemInfo messagingSystemInfo = oneAgentSDK.createMessagingSystemInfo("myMessagingSystem",
+		"requestQueue", MessageDestinationType.QUEUE, ChannelType.TCP_IP, "localhost:4711");
+OutgoingMessageTracer outgoingMessageTracer = oneAgentSDK.traceOutgoingMessage(messagingSystemInfo);
+outgoingMessageTracer.start();
+try {
+	// transport the dynatrace tag along with the message: 	
+	messageToSend.setHeaderField(
+		OneAgentSDK.DYNATRACE_MESSAGING_HEADERNAME, outgoingMessageTracer.getDynatraceStringTag());
+	theQueue.send(messageToSend);
+	
+	// optional:  add messageid provided from messaging system
+	outgoingMessageTracer.setVendorMessageId(toSend.getMessageId());
+	// optional:  add correlationId
+	outgoingMessageTracer.setCorrelationId(toSend.correlationId);
+} catch (Exception e) {
+	outgoingMessageTracer.error(e.getMessage());
+	Logger.logError(e);
+} finally {
+	outgoingMessageTracer.end();
+}
+```
+
+On the incoming side, we need to differ between the blocking receiving part and processing the received message. Therefore two different tracers are being used: ReceivingMessageTracer and ProcessingMessageTracer.
+
+```Java
+MessagingSystemInfo messagingSystemInfo = oneAgentSDK.createMessagingSystemInfo("myMessagingSystem",
+		"requestQueue", MessageDestinationType.QUEUE, ChannelType.TCP_IP, "localhost:4711");
+
+// following code is probably inside a busy loop:
```

Comment:
comment looks strange, a `while (true)` look should be inside a busy loop?

Reply:
also a loop that contains a blocking call isn't usually called a busy loop - especially when the blocking call doesn't have a timeout :)